libc-top-half/musl/src/thread/pthread_create.c

```diff
@@ -211,6 +240,24 @@ static int start_c11(void *p)
 	__pthread_exit((void *)(uintptr_t)start(args->start_arg));
 	return 0;
 }
+#else
+__attribute__((export_name("wasi_thread_start")))
+int wasi_thread_start(int tid, void *p)
+{
+	struct start_args *args = p;
+	// Set the thread ID on the pthread structure and save this structure as
+	// the global `pthread_self`.
+	args->thread->tid = tid;
+	__asm__("local.set %0\n"
+		  "global.set __wasilibc_pthread_self\n"
+          : "=r"(args->thread));
```

Comment:
Perhaps there is a better way to set this global?

Reply:
Currently the only way to set a real wasm global is via assembly (either .s file or inline like this).